app/controllers/metrics_controller.rb

```diff
@@ -11,13 +12,25 @@ class MetricsController < ApplicationController
   def index
     @position = POSITIONS[params[:position]]
     if @position == 'Quarterback'
-      @nfl_players = NflPlayer.joins(:career_passing_stats).where(position: @position, career_passing_stats: { year: 2015 }).limit(30).order('passing_yards DESC')
+      @nfl_players = NflPlayer.joins(:career_passing_stats).where(position: @position, career_passing_stats: { year: 2015 }).limit(30).order(sort_column + " " + "DESC")
     elsif @position == 'Running Back'
-      @nfl_players = NflPlayer.joins(:career_rushing_stats).where(position: @position, career_rushing_stats: { year: 2015 }).limit(50).order('rushing_yards DESC')
+      @nfl_players = NflPlayer.joins(:career_rushing_stats).where(position: @position, career_rushing_stats: { year: 2015 }).limit(50).order(sort_column + " " + "DESC")
     elsif @position == 'Wide Receiver'
-      @nfl_players = NflPlayer.joins(:career_receiving_stats).where(position: @position, career_receiving_stats: { year: 2015 }).limit(50).order('receiving_yards DESC')
+      @nfl_players = NflPlayer.joins(:career_receiving_stats).where(position: @position, career_receiving_stats: { year: 2015 }).limit(50).order(sort_column + " " + "DESC")
     elsif @position == 'Tight End'
-      @nfl_players = NflPlayer.joins(:career_receiving_stats).where(position: @position, career_receiving_stats: { year: 2015 }).limit(30).order('receiving_yards DESC')
+      @nfl_players = NflPlayer.joins(:career_receiving_stats).where(position: @position, career_receiving_stats: { year: 2015 }).limit(30).order(sort_column + " " + "DESC")
+    end
+  end
+
+  private
+
+  def sort_column
+    if @position == 'Quarterback'
+      CareerPassingStat.column_names.include?(params[:sort]) ? params[:sort] : "passing_yards"
+    elsif @position == 'Running Back'
+      CareerRushingStat.column_names.include?(params[:sort]) ? params[:sort] : "rushing_yards"
+    elsif @position == 'Wide Receiver' || 'Tight End'
+      CareerReceivingStat.column_names.include?(params[:sort]) ? params[:sort] : "receiving_yards"     
     end
```

Comment:
Note the similarity of the if/elseif blocks in the `index` and `sort_column` methods.

They both use String comparison of the `@position` variable to determine what to do. This pattern crops up regularly in software and is a sign to start considering introducing a new class and subclasses to take the place of the String variable and the `if/elsif` blocks. The new classes would simplify the code in the controller.

In this case, *start* thinking about introducing a `Position` class with subclasses of `Quarterback`, `RunningBack`, `WideReceiver`, and `TightEnd`. (There's no need to do this work yet, its enough to be aware of it as a future option).

Here's a rough draft of how that would look:

```ruby
class MetricsController ...
  ...
  def index
    @position = Position.lookup(params[:position])
    @nfl_players = @position.find_players(params[:sort])
  end
end
```


Note `Position` and subclasses are not database-backed models, as they do not subclass `ActiveRecord::Base`, they are just Plain Old Ruby Objects (sometimes called POROs for short):

```ruby
# File app/models/position.rb
class Position

  POSITIONS = {
    'quarterbacks' => Quarterback.new,
    'running-backs' => RunningBack.new,
    'wide-receivers' => WideReceiver.new,
    'tight-ends' => TightEnd.new
  }

  def self.lookup(position)
    POSITIONS[position]
  end

end
```

```ruby
# File app/models/quarterback.rb
class Quarterback < Position

  def find_players(sort)
    sort_column = CareerPassingStat.column_names.include?(sort) ? sort : "passing_yards"
    NflPlayer
      .joins(:career_passing_stats)
      .where(position: @position, career_passing_stats: { year: 2015 })
      .limit(30)
      .order(sort_column => :desc)
  end

  # to_partial_path will allow you to change the line in metrics/index.html.erb:
  # <%= render partial: @position.gsub(' ', '_').downcase, locals: { nfl_players: @nfl_players } %>
  # to:
  # <%= render @position, locals: { nfl_players: @nfl_players } %>
  # See: http://api.rubyonrails.org/classes/ActiveModel/Conversion.html#method-i-to_partial_path
  def to_partial_path
    'quarterback'
  end

end
```

```ruby
# File app/models/running_back.rb
class RunningBack < Position

  def self.find_players(sort)
    ...
  end

  def to_partial_path
    'running_back'
  end

end
```

and the other 2 classes for the other positions too.

This practice of replacing a String variable with objects is a refactoring pattern that can help simplify code and is one of the most commonly-employed techniques for improving the Object Oriented Design (OOD) of an application.